### DISCUSSION TITLE
Unclear error when doing auto-regression with too many forecasts and lags.

### DISCUSSION THREAD

**Discussion by pzaika** [ourownstory/neural_prophet#1257] [Feature Requests & Feedback]
When using auto-regression with a dataset that is smaller than the requested number of forecasts + lags (specifically when n_samples = len(df) - max_lags + 1 - n_forecasts is not positive) the fit method throws an IndexError. This hides the real problem, which is that the user is trying to train a model on not enough data. I think that it would be better to raise an error which tells the user to lower the number of forecasts or lags they are asking for.

**Comment by LeonieFreisinger** [comment]
Thanks for pointing that out. We will check back on it (see new issue)

**Comment by judussoari** [comment]
Solved with PR #1300 
Thanks for reporting this @pzaika
